### PR TITLE
[BE] [MPS] Validate random ops by metadata + stats

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.12.0) (oldest at bottom):
* #182386
* __->__ #182881

`test_output_match` and `test_output_grad_match` previously ran every op through `assertEqual` against the CPU reference, which is structurally meaningless for random ops: MPS and CPU consume independent RNG streams and rely on different RNG algorithms, so even seeded calls produce different draws. The pile of `"normal" / "uniform" / "rand_like" / "randn_like" / "randint" / "bernoulli" / "exponential" / "log_normal" / "cauchy" / "geometric" / "multinomial" / "nn.functional.dropout*"` entries in `UNDEFINED_XFAILLIST` existed solely to suppress these guaranteed-to-fail comparisons - the tests then validated nothing.

Adds `RANDOM_OP_NAMES` listing those ops and a `_assert_random_op_match` helper that  compares shape, dtype and layout

Removes the corresponding entries from `UNDEFINED_XFAILLIST` and `XFAILLIST_GRAD`. `randint(to>1, dtype=bool)` errors at the CPU op itself with "to - 1 is out of bounds for bool" - unrelated to the comparator - so its bool xfail entry is restored. The dropout family also keeps its grad-leg xfail, because backward reuses the forward random mask and MPS/CPU masks diverge by construction.

Spotted problem with MPSGraph normal implementation, when it did not work when mean and std shapes were broadcastable to each other.

Authored with Claude.

### CODE REVIEW

File: test/test_mps.py [hunk 1]
```diff
@@ -13484,6 +13484,61 @@ class TestConsistency(TestCaseMPS):
...
+        m_mps = _to_compare(mps_out)
+        m_cpu = _to_compare(cpu_out)
+        # Compare (min, max, mean) as a stat vector. atol scales with the
+        # draw's magnitude so e.g. exponential(rate=0.001) (mean ~1000) isn't
+        # held to a sub-unit slop; the rtol absorbs O(1/sqrt(N)) sampling
+        # noise. Goal is catching gross bugs, not validating distributions.
+        stats_mps = torch.stack([m_mps.min(), m_mps.max(), m_mps.mean()])
+        stats_cpu = torch.stack([m_cpu.min(), m_cpu.max(), m_cpu.mean()])
+        scale = max(m_cpu.abs().mean().item(), 1.0)
+        self.assertEqual(stats_mps, stats_cpu, atol=0.5 * scale, rtol=0.5)
```
Comment (manuelcandales) [inline test/test_mps.py]: I think the scale should involve the standard deviation of the original random variable

Comment (malfet) [reply test/test_mps.py]: You don't know it at the time you are running the test (because you can get there with random, or bernoulli or something), but you stats should be roughly the same

Comment (malfet) [reply test/test_mps.py]: But may be I should just compare metadata, and ignore everything else...

File: test/test_mps.py [hunk 2]
```diff
@@ -13484,6 +13484,61 @@ class TestConsistency(TestCaseMPS):
...
+        'nn.functional.feature_alpha_dropout',
+    }
+
+    def _assert_random_op_match(self, mps_out, cpu_out):
+        # Strict metadata: shape / dtype / layout must match exactly.
+        self.assertEqual(mps_out.shape, cpu_out.shape)
+        self.assertEqual(mps_out.dtype, cpu_out.dtype)
+        self.assertEqual(mps_out.layout, cpu_out.layout)
+        # Loose statistics: with N >= 100 IID samples from the same distribution
+        # both means should agree to ~O(1/sqrt(N)) and the supports should
```
Comment (manuelcandales) [inline test/test_mps.py]: Isn't this missing the standard deviation of the distribution. i.e. the standard deviation of the sample mean is the original standard deviation divided by sqrt(N).

File: test/test_mps.py (hunk 1 above)
Comment (manuelcandales) [reply test/test_mps.py]: I am worried about the line that says: "both means should agree to ~O(1/sqrt(N))". I think this is false. If the original standard deviation is large, this won't hold, because the standard deviation of the sample mean depends on the original standard deviation (i.e. it equals the original standard deviation divided by sqrt(N)). I think Claude is assuming a distribution with standard deviation = 1.

**Review by manuelcandales — APPROVED** [review]
stamping to unblock

File: test/test_mps.py (hunk 2 above)
Comment (malfet) [reply test/test_mps.py]: I've removed everything to just focus on metadata checks, as it's better than xfail.... will add checks later
